### PR TITLE
chore: add details metric for anchor write

### PR DESCRIPTION
**Pull request by fqutishat**
Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: pkg/metrics/metrics.go [hunk 1]
```diff
@@ -231,6 +245,34 @@ func (m *Metrics) WriteAnchorTime(value time.Duration) {
...
+	m.anchorWriteBuildCredTime.Observe(value.Seconds())
+
+	logger.Infof("WriteAnchor build credential time: %s", value)
+}
+
+// WriteAnchorGetWitnessesTime records the time it takes to get witnesses inside write anchor.
+func (m *Metrics) WriteAnchorGetWitnessesTime(value time.Duration) {
+	m.anchorWriteGetWitnessesTime.Observe(value.Seconds())
+
+	logger.Infof("WriteAnchor get witness time: %s", value)
```
Comment (bstasyszyn) [inline pkg/metrics/metrics.go:259]: I would make this a debug log.

File: pkg/metrics/metrics.go [hunk 2]
```diff
@@ -231,6 +245,34 @@ func (m *Metrics) WriteAnchorTime(value time.Duration) {
 	logger.Infof("WriteAnchor time: %s", value)
 }
 
+// WriteAnchorBuildCredentialTime records the time it takes to build credential inside write anchor.
+func (m *Metrics) WriteAnchorBuildCredentialTime(value time.Duration) {
+	m.anchorWriteBuildCredTime.Observe(value.Seconds())
+
+	logger.Infof("WriteAnchor build credential time: %s", value)
```
Comment (bstasyszyn) [inline pkg/metrics/metrics.go:252]: I would make this a debug log. We already have an Info for the whole WriteAnchor time (maybe that should also be a debug).

File: pkg/metrics/metrics.go [hunk 3]
```diff
@@ -231,6 +245,34 @@ func (m *Metrics) WriteAnchorTime(value time.Duration) {
...
+	m.anchorWriteGetWitnessesTime.Observe(value.Seconds())
+
+	logger.Infof("WriteAnchor get witness time: %s", value)
+}
+
+// WriteAnchorSignCredentialTime records the time it takes to sign credential inside write anchor.
+func (m *Metrics) WriteAnchorSignCredentialTime(value time.Duration) {
+	m.anchorWriteSignCredTime.Observe(value.Seconds())
+
+	logger.Infof("WriteAnchor sign credential time: %s", value)
```
Comment (bstasyszyn) [inline pkg/metrics/metrics.go:266]: I would make this a debug log.

File: pkg/metrics/metrics.go [hunk 4]
```diff
@@ -231,6 +245,34 @@ func (m *Metrics) WriteAnchorTime(value time.Duration) {
...
+}
+
+// WriteAnchorPostOfferActivityTime records the time it takes to post offer activity inside write anchor.
+func (m *Metrics) WriteAnchorPostOfferActivityTime(value time.Duration) {
+	m.anchorWritePostOfferActivityTime.Observe(value.Seconds())
+
+	logger.Infof("WriteAnchor sign credential time: %s", value)
```
Comment (bstasyszyn) [inline pkg/metrics/metrics.go:273]: I would make this a debug log.

File: pkg/metrics/metrics.go (hunk 1 above)
Comment (fqutishat) [reply pkg/metrics/metrics.go]: done

File: pkg/metrics/metrics.go (hunk 2 above)
Comment (fqutishat) [reply pkg/metrics/metrics.go]: done

File: pkg/metrics/metrics.go (hunk 3 above)
Comment (fqutishat) [reply pkg/metrics/metrics.go]: done

File: pkg/metrics/metrics.go (hunk 4 above)
Comment (fqutishat) [reply pkg/metrics/metrics.go]: done

**Review by bstasyszyn — APPROVED** [review]

**Review by sandrask — APPROVED** [review]

**Pull request merged**